src/mlkem.rs

```diff
@@ -268,6 +273,21 @@ impl Mechanism for MlKemMechanism {
         &self.info
     }
 
+    /// Get expected length of the encapsulated ciphertext for given key
+    ///
+    /// Returns the size in bytes or Err if the key is not ML-KEM
+    fn encapsulate_ciphertext_len(&self, key: &Object) -> Result<usize> {
+        match key.get_attr_as_ulong(CKA_PARAMETER_SET) {
```

Comment:
You should check that CKA_KEY_TYPE matches CKK_ML_KEM first.

ML-DSA/SLH-DSA also have CKA_PARAMETER_SET and the CKP values overlap, so this function may not error out on the wrong key type.

Reply:
Good point! I thought this is already done in the C_Encapsulate, but it is only done in the `encapsulate`. Done.